llms/mlx_lm/utils.py

```diff
@@ -143,17 +143,21 @@ def sample(logits: mx.array) -> Tuple[mx.array, float]:
         if temp == 0:
             token = mx.argmax(logits, axis=-1)
         else:
+            # referenced implementation from https://github.com/huggingface/transformers/blob/main/src/transformers/generation/logits_process.py#L449-L460
             if top_p > 0 and top_p < 1.0:
                 if (
                     logits.dtype == mx.bfloat16
                 ):  # workdaround for unable to load kernel contiguous_scan_inclusive_sum_bfloat16_bfloat16
                     logits = logits.astype(mx.float32)
+
                 probs = mx.softmax(logits / temp, axis=-1)
 
-                sorted_probs = mx.sort(probs)[::-1]
-                sorted_indices = mx.argsort(probs)[::-1]
+                # sort probs in ascending order
+                sorted_probs = mx.sort(probs, axis=-1)
```

Comment:
I am not sure if the MLX sort implementation changes or not, but I did some tests today and it seems that the MLX sort returns the array in ascending order.

Reply:
It has not changed, it's always been ascending (following numpy)

Reply:
Cool, so the original implementation was wrong then, it must have been mixed up with the PyTorch sort implementation.